packages/query/tests/query-replication-state.test.ts

```
@@ -42,3 +42,83 @@ describe('QueryReplicationState', () => {
+		expect(httpClient.post).toHaveBeenCalledTimes(1);
+		await replicationState.cancel();
+	});
+
+	it('keeps rerunning after progress even when unsynced ids do not shrink immediately', async () => {
+		const { httpClient, replicationState } = createGreedyState({
+			unsyncedRemoteIDs: [
+				[1, 2, 3],
+				[1, 2, 3],
+				[1, 2, 3],
+				[],
+			],
```

ESLint

[error] Replace `⏎↹↹↹↹[1,·2,·3],⏎↹↹↹↹[1,·2,·3],⏎↹↹↹↹[1,·2,·3],⏎↹↹↹↹[],⏎↹↹↹` with `[1,·2,·3],·[1,·2,·3],·[1,·2,·3],·[]`

(prettier/prettier)